website/docs/guides/advanced-cluster-new-sharding-schema.html.markdown

```diff
@@ -0,0 +1,344 @@
+      provider_name = "AWS"
+      priority      = 7
+      region_name   = "EU_WEST_1"
+    }
+  }
+}
+```
+
+This  updated configuration will trigger a terraform update plan, however the underlying cluster will not face any changes after the apply as both configurations represent a sharded cluster composed of 2 shards.
+
+<a id="migration-geosharded"></a>
+## Migrating existing advanced_cluster type GEOSHARDED
+
+Considering the following configuration of a GEOSHARDED cluster using the deprecated `num_shards`:
+
+```
+resource "mongodbatlas_advanced_cluster" "test" {
+  project_id = var.project_id
+  name = "GeoShardedCluster"
+  cluster_type   = "GEOSHARDED"
+
+  replication_specs {
+    zone_name  = "zone n1"
+    num_shards = 2
+    region_configs {
+    electable_specs {
+        instance_size = "M10"
+        node_count    = 3
+    }
+    provider_name = "AWS"
+    priority      = 7
+    region_name   = "US_EAST_1"
+    }
+  }
+
+  replication_specs {
+    zone_name  = "zone n2"
+    num_shards = 2
+
+    region_configs {
+    electable_specs {
+        instance_size = "M10"
+        node_count    = 3
+    }
+    provider_name = "AWS"
+    priority      = 7
+    region_name   = "EU_WEST_1"
+    }
+  }
+}
+```
+
+In order to update our configuration to the new schema we will remove the use of `num_shards` and add a new identical `replication_specs` element for each shard. Note that these 2 changes must be done at the same time.
+
+```
+resource "mongodbatlas_advanced_cluster" "test" {
+  project_id = var.project_id
+  name = "GeoShardedCluster"
+  cluster_type   = "GEOSHARDED"
+
+  replication_specs { # first shard for zone n1
+    zone_name  = "zone n1"
+    region_configs {
+    electable_specs {
+        instance_size = "M10"
+        node_count    = 3
+    }
+    provider_name = "AWS"
+    priority      = 7
+    region_name   = "US_EAST_1"
+    }
+  }
+
+  replication_specs { # second shard for zone n1
+    zone_name  = "zone n1"
+    region_configs {
+    electable_specs {
+        instance_size = "M10"
+        node_count    = 3
+    }
+    provider_name = "AWS"
+    priority      = 7
+    region_name   = "US_EAST_1"
+    }
+  }
+
+  replication_specs { # first shard for zone n2
+    zone_name  = "zone n2"
+    region_configs {
+    electable_specs {
+        instance_size = "M10"
+        node_count    = 3
+    }
+    provider_name = "AWS"
+    priority      = 7
+    region_name   = "EU_WEST_1"
+    }
+  }
+
+  replication_specs { # second shard for zone n2
+    zone_name  = "zone n2"
+    region_configs {
+    electable_specs {
+        instance_size = "M10"
+        node_count    = 3
+    }
+    provider_name = "AWS"
+    priority      = 7
+    region_name   = "EU_WEST_1"
+    }
+  }
+}
+```
+
+
+
+This  updated configuration will trigger a terraform update plan, however the underlying cluster will not face any changes after the apply as both configurations represent a geo sharded cluster with 2 zones and 2 shards in each one.
```

Comment:
same as above

Reply:
Suggested rephrasing:

This updated configuration will trigger a `terraform update plan`. However, the underlying cluster will not change after the apply, as both configurations represent a geo-sharded cluster with two shards in each of the two zones that the cluster spans.